services/src/datasets/external/pangaea/meta.rs

```diff
@@ -0,0 +1,624 @@
+            }
+        }
+        None
+    }
+
+    /// Creates a new tsv header matching the actual column names.
+    /// The new header also includes a column for the default geometry,
+    /// if applicable.
+    fn tsv_header(&self) -> String {
+        let mut result = String::new();
+
+        for (idx, p) in self.columns.iter().enumerate() {
+            if idx > 0 {
+                result.push('\t');
+            }
+            result.push_str(p.full_name().as_str());
+        }
+
+        // Default geometry
+        match &self.feature_info {
+            FeatureInfo::DefaultPoint { .. } | FeatureInfo::DefaultPolygon { .. } => {
+                result.push('\t');
+                result.push_str(DEFAULT_FEATURE_COLUMN_NAME);
+            }
+            _ => {}
+        };
+        result
+    }
+
+    fn get_result_descriptor(&self) -> VectorResultDescriptor {
+        let feature_type = match &self.feature_info {
+            FeatureInfo::None => VectorDataType::Data,
+            FeatureInfo::DefaultPolygon { .. } => VectorDataType::MultiPolygon,
+            _ => VectorDataType::MultiPoint,
+        };
+
+        let column_map: HashMap<String, FeatureDataType> = self
+            .columns
+            .iter()
+            .map(|param| match param {
+                PangeaParam::Numeric { name: n, .. } => (n.clone(), FeatureDataType::Float),
+                PangeaParam::String { name: n, .. } => (n.clone(), FeatureDataType::Text),
+            })
+            .collect();
+
+        VectorResultDescriptor {
+            spatial_reference: SpatialReference::epsg_4326().into(),
+            data_type: feature_type,
+            columns: column_map,
+        }
+    }
+
+    fn get_column_spec(&self) -> OgrSourceColumnSpec {
+        let (x, y) = match &self.feature_info {
+            FeatureInfo::Point {
+                lat_column: y,
+                lon_column: x,
+            } => (x.clone(), Some(y.clone())),
+            FeatureInfo::DefaultPoint { .. } | FeatureInfo::DefaultPolygon { .. } => {
+                (DEFAULT_FEATURE_COLUMN_NAME.to_owned(), None)
+            }
+            FeatureInfo::None => ("".to_string(), None),
+        };
+
+        OgrSourceColumnSpec {
+            x,
+            y,
+            rename: None,
+            float: self
+                .columns
+                .iter()
+                .filter(|&c| matches!(c, PangeaParam::Numeric { .. }))
+                .map(PangeaParam::full_name)
+                .collect(),
+            int: vec![],
+            text: self
+                .columns
+                .iter()
+                .filter(|&c| matches!(c, PangeaParam::String { .. }))
+                .map(PangeaParam::full_name)
+                .collect(),
+        }
+    }
+
+    fn get_ogr_source_ds(&self, memfile_name: &str) -> OgrSourceDataset {
+        let path: PathBuf = PathBuf::from(memfile_name);
+        let layer = path.file_stem().unwrap().to_str().unwrap().to_string();
```

Comment:
I guess it is better to not unwrap but throw an error.
If you are 100% sure that it won't panic, you can use expect.

Otherwise, our threads will die at some point.

Please also look at the other unwraps in this file (tests are fine, but the ones in regular functions)

Reply:
4fe6e64bb653a4451d5a92e4ef1d66e9b22952e9